### PR TITLE
Add branch-protection configuration for test-infra

### PR DESCRIPTION
**Pull request by fejta**
/assign @stevekuznetsov @spiffxp @timothysc 

ref https://github.com/kubernetes/test-infra/pull/6205

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
/lgtm

/hold

In case there are any considerations before rolling it out.

File: prow/config.yaml [hunk 1]
```diff
@@ -22,6 +22,15 @@ prowjob_namespace: default
 pod_namespace: test-pods
 log_level: info
 
+branch-protection:
+  Orgs:
```
Comment (cblecker) [inline prow/config.yaml]: I know this is correct per the code, but the capitalization of Orgs and Repos looks odd in the configuration, as there is no other key in this file that begins with a capital letter.

Comment (fejta) [reply prow/config.yaml]: Oh good catch! Actually I think it is supposed to be wrong per the code?

https://github.com/kubernetes/test-infra/blob/master/prow/config/branch_protection.go

Comment (cblecker) [reply prow/config.yaml]: You are absolutely right! I was looking at the wrong place.

**Pull request merged**